lib/icinga/checkable-notification.cpp

```diff
@@ -145,35 +146,56 @@ static void FireSuppressedNotifications(Checkable* checkable)
 
 	int subtract = 0;
 
-	for (auto type : {NotificationProblem, NotificationRecovery, NotificationFlappingStart, NotificationFlappingEnd}) {
-		if (suppressed_types & type) {
-			bool still_applies = checkable->NotificationReasonApplies(type);
-
-			if (still_applies) {
-				bool still_suppressed;
-
-				switch (type) {
-					case NotificationProblem:
-						/* Fall through. */
-					case NotificationRecovery:
-						still_suppressed = !checkable->IsReachable(DependencyNotification) || checkable->IsInDowntime() || checkable->IsAcknowledged();
-						break;
-					case NotificationFlappingStart:
-						/* Fall through. */
-					case NotificationFlappingEnd:
-						still_suppressed = checkable->IsInDowntime();
-						break;
-					default:
-						break;
-				}
+	{
+		int_fast8_t wasLastParentRecoveryRecent = -1;
```

Comment:
So if I understand correctly, this is supposed to be a bool plus an additional state "not computed yet" (-1). Maybe use `boost::optional<bool>` instead?